functions-and-scope.js

```diff
@@ -0,0 +1,199 @@
+// maar ook een manier moeten vinden om hetgeen dat je verzamelt ergens te bundelen. Op deze manier zul je ontdekken hoe je omgaat met scope. Pak vooral het hoofdstuk op EdHub over for-loops er nog eens bij!
+// Tip: je mag hier geen ingebouwde object methoden gebruiken, dus daar hoef je niet naar te kijken.
+
+const grades = [9, 8, 5, 7, 7, 4, 9, 8, 8, 3, 6, 8, 5, 6];
+/* Opdracht  1: Cum Laude */
+
+/* 1a: Script schrijven  */
+// De administratie moet weten hoeveel studenten er dit blok cum laude zijn afgestudeerd (8 of hoger). Daar moeten namelijk speciale diploma's voor besteld worden.
+// Schrijf de stapjes om dit te kunnen checken eerst uit en vraag jezelf de volgende dingen af:
+// * Hoe kan ik iedere waarde van de array checken op deze conditie?
+// * Hoe zorg ik ervoor dat dit ook werkt wanneer de array 100 entries bevat?
+// * Hoe zorgt ik ervoor dat wanneer ik een cijfer tegenkom die aan de conditie voldoet, ik dit ergens kan bijhouden?
+// Log het antwoord in de terminal.
+
+// ---- Verwachte uitkomst: 6
+let outcome = 0
+
+for (let i = 0; i < grades.length; i++) {
+    if( grades[i] >= 8) {
+        outcome = outcome + 1
+    }
+}
+
+console.log(outcome)
+
+
+/*  1b: Omschrijven tot een herbruikbare functie   */
+// Schrijf een functie genaamd cumLaude, die een array van cijfers verwacht (zoals grades) en het aantal Cum laude studenten teruggeeft. Gebruik hiervoor jouw antwoord van 1a.
+// Zorg ervoor dat jouw functie ook werkt als we een andere array met eindcijfers willen checken, zoals bijvoorbeeld: [6, 4, 5] of [8, 9, 4, 6, 10].
+// Log het antwoord in de terminal.
+
+
+let cumlaudeOutcome = 0;
+function cumlaude (array){
+    for (let i = 0; i < array.length; i++) {
+        if( array[i] >= 8) {
+            cumlaudeOutcome = cumlaudeOutcome + 1
+        }
+    }
+    return cumlaudeOutcome
+}
+const outcome2 = cumlaude([8, 9, 4, 6, 10])
+console.log(outcome2)
+
+// ---- Verwachte uitkomsten:
+// cumLaude(grades) geeft 6
+// cumLaude([6, 4, 5]) geeft 0
+// cumLaude([8, 9, 4, 6, 10]) geeft 3
+
+
+/* Opdracht  2: Gemiddeld cijfer */
+
+/* 2a: Script schrijven  */
+// De studenten-administratie moet ieder blok opnieuw berekenen wat het gemiddelde eindcijfer is, maar we beginnen met de grades array van hierboven.
+// Schrijf de stapjes om dit te kunnen berekenen eerst uit en vraag jezelf de volgende dingen af:
+// * Hoe wordt een gemiddelde berekend?
+
+//alles bij elkaar opgeteld en dan door elkaar gedeeld
+
+// * Wat moet ik verzamelen uit de array van cijfers om uiteindelijk een gemiddelde te kunnen berekenen?
+
+//alle cijfers bij elkaar gedeeld door the length
+
+// * Hoe zorgt ik ervoor dat ik alle waardes uit de array kan langslopen, ook als de array wel 100 entries zou bevatten?
+// Log het antwoord in de terminal.
+
+let averageMark = 0
+for (let i = 0; i < grades.length; i++) {
+    if ([i] < grades.length -1 ) {
```

Comment:
Er was hier geen if of else statement nodig, we konden simpel weg in de for-loop alle waardes optellen:
`averageMark = averageMark + grades[i];`

en vervolgens in de console.log:
`console.log(averageMark / grades.length);`